### PR TITLE
MRG: Textures

### PR DESCRIPTION
**Pull request by Eric89GXL**
This adds support for temporal coherence textures. WIP because I need to add tests.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
@drammock ready for review/merge from my end

File: examples/stimuli/plot_texture_stimuli.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+fig, ax = plt.subplots(1, figsize=(8, 2))
+img = ax.specgram(sig, NFFT=1024, Fs=fs, noverlap=800)[3]
+img.set_clim([img.get_clim()[1] - 50, img.get_clim()[1]])
+ax.set(xlim=t[[0, -1]], ylim=[0, 10000], xlabel='Time (sec)',
+       ylabel='Freq (Hz)')
+fig.tight_layout()
+
+###############################################################################
+# And the long-term spectrum:
+
```
Comment (drammock) [inline examples/stimuli/plot_texture_stimuli.py]: extra line break?

File: expyfun/stimuli/_texture.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        ``int(np.round(n_freqs * 0.8))``.
+    rho : float
+        Correlation between the envelopes of grouped tones (default is 1.0).
+    seq : list
+        Sequence of incoherent ('inc'), coherent noise envelope ('nb'), and
+        SAM ('sam') mixtures. Default is ``('inc', 'nb', 'inc', 'nb')``.
+    fs : float
+        Sampling rate in Hz.
+    dur : float
+        Duration (in seconds) of each token in seq (default is 1.0).
```
Comment (drammock) [inline expyfun/stimuli/_texture.py]: might be nice for this to accept a sequence of durations, like if you wanted the `inc` to be 1.5 s followed by `nb` of 0.5 s.  But that can be a separate PR later, on YAGNI grounds.

File: expyfun/stimuli/_texture.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        The SAM frequency to use.
+    freq_lims : tuple
+        The lower and upper frequency limits (default is (200, 8000)).
+    random_state : None | int | np.random.RandomState
+        The random generator state used for band selection and noise
+        envelope generation.
+
+    Returns
+    -------
+    x : ndarray, shape (N,)
```
Comment (drammock) [inline expyfun/stimuli/_texture.py]: shouldn't shape be `(n_freqs, len(seq) * dur * fs)` ?

File: expyfun/stimuli/_texture.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        The stimulus.
+
+    Notes
+    -----
+    This function requires MNE.
```
Comment (drammock) [inline expyfun/stimuli/_texture.py]: do we have a `@requires_mne` decorator?  Does it make sense to add one?

Comment (Eric89GXL) [inline expyfun/stimuli/_texture.py]: YAGNI (until we expand to users who don't use MNE)

File: expyfun/stimuli/_texture.py (hunk 3 above)
Comment (Eric89GXL) [inline expyfun/stimuli/_texture.py]: no `n_freqs` but otherwise, yes

File: expyfun/stimuli/_texture.py (hunk 2 above)
Comment (Eric89GXL) [inline expyfun/stimuli/_texture.py]: Agreed YAGNI until we hit a use case

**Comment by Eric89GXL** [comment]
Thanks @drammock changes pushed. I don't think @rkmaddox or @haribharadwaj will likely have time to look at this one (but feel free to prove me wrong!). But FWIW I based this on Hari's code and have bothered him with questions about it already during implementation, so that's a good start.

**Pull request merged**